src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,220 @@
+                + " manufacturers.country as m_c from cars"
+                + " inner join cars_drivers on cars.id = cars_drivers.car_id"
+                + " inner join drivers on drivers.id = cars_drivers.driver_id"
+                + " inner join manufacturers on manufacturers.id = cars.manufacturer_id"
+                + " where drivers.id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCarsFromDB(resultSet));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private Car getCarFromDB(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        Manufacturer manufacturer = new Manufacturer();
+        resultSet.beforeFirst();
+        if (resultSet.next()) {
+            manufacturer.setId(resultSet.getObject("c_mid", Long.class));
+            manufacturer.setName(resultSet.getObject("m_n", String.class));
+            manufacturer.setCountry(resultSet.getObject("m_c", String.class));
+            car.setId(resultSet.getObject("c_id", Long.class));
+            car.setModel(resultSet.getObject("c_md", String.class));
+            car.setManufacturer(manufacturer);
+        }
+        while (resultSet.next()) {
+            Driver driver = new Driver();
+            driver.setId(resultSet.getObject("d_id", Long.class));
+            driver.setName(resultSet.getObject("d_n", String.class));
+            driver.setLicenceNumber(resultSet.getObject("d_ln", String.class));
+            car.setDrivers(driver);
+        }
+        return car;
+    }
+
+    private void createCarsDriversDB(Car car) {
+        String queryForCarsDrivers = "INSERT INTO cars_drivers (driver_id, car_id) values (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            Iterator<Driver> iterator = car.getDrivers().iterator();
+            while (iterator.hasNext()) {
+                preparedStatementOfCarsDrivers.setLong(1, iterator.next().getId());
+                preparedStatementOfCarsDrivers.setLong(2, car.getId());
+                preparedStatementOfCarsDrivers.executeUpdate();
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create data "
+                    + car + " into cars_drivers table", exception);
+        }
+    }
+
+    private void deleteDataFromCarsDrivers(Connection connection, Car car) {
```

Comment:
```suggestion
    private void deleteDataFromCarsDrivers(Car car) {
```
create a separate connection here